src/utils/process.c

```diff
@@ -51,8 +51,30 @@ int spawn_picomling(struct x_connection *c) {
 	setsid();
 
 	x_connection_init_xcb(c, new_c, screen);
-	xcb_prefetch_extension_data(c->c, &xcb_render_id);
+
 	xcb_prefetch_extension_data(c->c, &xcb_randr_id);
+	xcb_prefetch_extension_data(c->c, &xcb_render_id);
+
+	// Initialize the X RandR extension.
+	auto extension = xcb_get_extension_data(c->c, &xcb_randr_id);
+	if (!extension || !extension->present) {
+		log_error("The X server doesn't have the X RandR extension.");
+
+		return -1;
+	}
+
+	c->e.has_randr = true;
+	c->e.randr_event = extension->first_event;
+
+	// Initialize the X Render extension.
+	extension = xcb_get_extension_data(c->c, &xcb_render_id);
+	if (!extension || !extension->present) {
+		log_error("The X server doesn't have the X Render extension.");
+
+		return -1;
+	}
+
+	c->e.render_error = extension->first_error;
```

Comment:
i'm not sure that it's possible for the information about the x extensions to differ among multiple x connection thus not sure that this is needed. afaiu the way a connection to the x server is made here reuses the main `x_connection` structure that already has the inner `x_extensions` structure filled as this function is called after the `x_extensions_init` function.

Reply:
i'd prefer you just call `x_extensions_init` here.

Reply:
does the current variant look better? it'll initialize all the x extensions used by picom even if they're not used by picomling but if it's fine then the pull request is ready to be merged.

Reply:
yes, it looks fine.